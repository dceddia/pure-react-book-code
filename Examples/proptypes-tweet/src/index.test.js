jest.mock('react-dom', () => ({render: jest.fn()}))

import React from 'react';
import { Tweet, testTweet } from './index';
import renderer from 'react-test-renderer';

it('renders', () => {
  const tree = renderer.create(
    <Tweet tweet={testTweet}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
