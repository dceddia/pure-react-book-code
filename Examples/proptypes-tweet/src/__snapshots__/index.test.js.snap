// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="tweet"
>
  <img
    alt="avatar"
    className="avatar"
    src="https://www.gravatar.com/avatar/xyz"
  />
  <div
    className="content"
  >
    <span
      className="author"
    >
      <span
        className="name"
      >
        IAMA Cat Person
      </span>
      <span
        className="handle"
      >
        @
        catperson
      </span>
    </span>
    <span
      className="time"
    >
      2 years ago
    </span>
    <div
      className="message"
    >
      Something about cats.
    </div>
    <div
      className="buttons"
    >
      <i
        className="fa fa-reply reply-button"
      />
      <span
        className="retweet-button"
      >
        <i
          className="fa fa-retweet"
        />
        <span
          className="retweet-count"
        >
          17
        </span>
      </span>
      <span
        className="like-button"
      >
        <i
          className="fa fa-heart"
        />
        <span
          className="like-count"
        >
          2
        </span>
      </span>
      <i
        className="fa fa-ellipsis-h more-options-button"
      />
    </div>
  </div>
</div>
`;
